dhnx/optimization/precalc_hydraulic.py

```diff
@@ -25,48 +25,345 @@
 #   - https://www.schweizer-fn.de/stroemung/rauhigkeit/rauhigkeit.php
 
 
-def delta_p(v, d_i, k=0.1, T_medium=90, l=1,
-            pressure=1, fluid='IF97::Water'):
+def eq_smooth(x, *R_e):
+    r"""
+    Equation to be solved for smooth surfaces
+
+    Equation
+    --------
+    .. eq_smooth_equation:
+
+    :math:`f(x) = x-2 \cdot log\Big(\frac{Re}{2,51x}\Big)`
+
+    Parameters
+    ----------
+    x: numeric
+        :math:`x`: function variable
+    R_e: numeric
+        :math:`Re`: Reynolds number
+
+    Returns
+    -------
+    Equation : numeric
+
+    """
+    return x - 2 * np.log10(R_e / (x * 2.51))
+    # für argumente siehe https://stackoverflow.com/questions/19843116/passing-arguments-to-fsolve
+
+
+def eq_transition(x, *data):
+    r"""
+    Equation to be solved for the transition range.
+
+    Equation
+    --------
+    .. eq_transition_equation
+
+    :math:`f(x)=x+2\cdot log \Bigg(  \frac{2,51x}{Re} \cdot \frac{k}{3,71d_i} \Bigg)`
+
+    Parameters
+    ----------
+    x : numeric
+        :math:`x`: function variable
+
+    R_e: numeric
+        :math:`Re`: Reynolds number
+
+    k : numeric
+        :math:`k`: roughness of inner pipeline surface [mm]
+
+    di : numeric
+        :math:`d_i`: inner diameter [m]
+
+    Returns
+    -------
+    Equation : numeric
+
+    """
+    R_e, k, d_i = data  # siehe oben
+
+    return x + 2 * np.log10((2.51 * x) / R_e + k / (3.71 * d_i))
+
+
+def calc_k_v(d_v, d):
+    r"""
+    Calcutlates the kinematic viscosity for given density and dynamic viscosity
+
+    Formula
+    -------
+    .. calc_k_v_equation:
+
+    :math:`\nu = \frac{\eta}{\rho}`
+
+    Parameters
+    ----------
+    d_v: numeric
+        :math:`\eta`: dynamic viskosity [kg/(m*s)]
+
+    d: numeric
+        :math:`\rho`: density [kg/m³]
+
+    Returns
+    -------
+    kinematic viscosity [m²/s] : numeric
+
+    """
+    return d_v / d
+
+
+def calc_Re(v, d_i, k_v):
+    r"""
+    Calculates the Reynolds number for a given velocity, inner diameter and kinematic viscosity
+
+    Formula
+    -------
+    .. calc_Re_equation:
+
+    :math:`\frac{v \cdot d_i}{\nu}`
+
+    Parameters
+    ----------
+    v: numeric
+        :math:`v`: flow velocity [m/s]
+
+    d_i: numeric
+        :math:`d_i`: inner pipe diameter [m]
+
+    k_v: numeric
+        :math:`\nu`: kinematic viscosity [m²/s]
+
+    Returns
+    -------
+    Reynolds number []: numeric
```

Comment:
```suggestion
    Reynolds number [-]: numeric
```

I am used to a `[-]` to show that its unitsless. What do you think?